auto_tuning/proxy/src/proxy_common.hpp

```diff
@@ -0,0 +1,117 @@
+#ifndef SEISSOL_PROXY_PROXY_COMMON_DATATYPES_HPP
+#define SEISSOL_PROXY_PROXY_COMMON_DATATYPES_HPP
+
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <stdexcept>
+
+enum Kernel {
+  all = 0,
+  local,
+  neigh,
+  ader,
+  localwoader,
+  neigh_dr,
+  godunov_dr
+};
+
+struct ProxyConfig {
+  unsigned cells{static_cast<unsigned>(1e5)};
+  unsigned timesteps{10};
+  Kernel kernel{Kernel::all};
+  bool verbose{true};
+};
+
+struct ProxyOutput{
+  double time{};
+  double cycles{};
+  double libxsmmNumTotalGFlop{};
+  double pspammNumTotalGFlop{};
+  double libxsmmAndpspammNumTotalGFlop{};
+  double actualNonZeroGFlop{};
+  double actualHardwareGFlop{};
+  double gib{};
+  double nonZeroFlopPerCycle{};
+  double hardwareFlopPerCycle{};
+  double bytesPerCycle{};
+  double nonZeroGFlops{};
+  double hardwareGFlops{};
+  double gibPerSecond{};
+};
+
+ProxyOutput runProxy(ProxyConfig config);
+
+struct Aux {
```

Comment:
Why is that a struct? It only has static members. Should be a namespace imo. Or is it needed for pybind?

Reply:
Yes, I can do it. I used `struct` as kinda namespace.

> Or is it needed for pybind?

No, you can bind methods as well

Reply:
I remembered why I did in this way. So, if it a struct it is going to be a used-derived type with a bunch of inline static methods and I can safely include it in other source files avoiding multiple definitions. If it is a bunch of functions then I need to create another translation unit and add it to cmake. In general, I would leave it as it is.

In terms of namespaces, I agree but the entire proxy was written without namespaces. I guess it is better to introduce proxy namespace in a separate PR.

Reply:
Okay, sure. Doesn't matter. There shouldn't be any problem when defining inline functions imultiple times. That's exactly what inline should allow.